modRest refactoring

Jako opened the pull request:
### What does it do?

Some code refactoring
### Why is it needed?
- The return values don't match the PHPDoc definitions and the usage was different too
- Wrong phpDoc type in recursive method call
- Removed unused variables
- Updated property descriptions
- Method names are case sensitive

Jako committed:
Refactored return values
Reason: The return values don't match the PHPDoc definitions and the usage was different too.

Jako committed:
Wrong parameter type in recursive method call
According to the code it should be an array too

Jako committed:
Removed unused variables

Jako committed:
Removed unused variables

Jako committed:
Updated property descriptions

Jako committed:
Method case sensivity

Jako committed:
Wrong phpDoc type in recursive method call

Mark-H commented on core/model/modx/rest/modrest.class.php:
```diff
@@ -473,7 +473,7 @@ public function toXml($parameters,$rootNode) {
      */
     protected function _populateXmlDoc(&$doc, &$node, &$parameters) {
-        /** @var $val DOMNode */
+        /** @var DOMnode/array $val */
```
I don't think that's the right format for phpdoc is it? Isn't that supposed to be `DOMnode[]`?

Jako replied:
Maybe `DOMnode[]/DOMnode`

Alroniks replied:
Alternatives should be separated using `|` instead of `/`. So correct way `DOMnode|array`

Jako replied:
$parameters has now the right phpdoc type. The type warnings are away now.

Mark-H reviewed (commented):
Could you elaborate on the reasoning behind changing from returning values to returning nothing? Sounds like a breaking change and I don't immediately see the benefit, so some more thoughts on what these changes accomplish would be great.

Jako commented:
The main reason for the change was that $this->success and $this->failure have no return values. The other methods use those return values. So the return was null every time (if the success and failure methods are not overridden).

pixelchutes commented:
Thanks @Jako, was curious about the reasoning there.

Jako committed:
Add status code and change the order of the response array
to get the same answer as when the API endpoint not exists

Jako committed:
Fixing Method not allowed for not numeric IDs
$id could be a string that does not start with a number. In that case the check with integer casting is not valid. It should be enough if $id is not empty.

Jako committed:
Allow to switch off the checkForSuffix test on base of a service setting.

Jako committed:
Right phpdoc type for $parameters

Mark-H commented:
Thanks @Jako!
